### DISCUSSION TITLE
Where are the derivative rules for conv_general_dilated

### DISCUSSION THREAD

**Discussion by Rasmuskh** [jax-ml/jax#12430] [General] [answered]
Hi,
I am trying to find the derivative rules for conv_general_dilated (and conv_general_dilated_local). 
Where in the souce code are these rules? or does jax just generate them at compile time somehow?

**Comment by hawkinsp** [comment, marked as answer]
You're looking for the `transpose` rules here: https://github.com/google/jax/blob/fc2902c6ac2e7b7f228e6637a3fc97698fc1b1cf/jax/_src/lax/convolution.py#L427

Convolution is linear in both its arguments, so we have only `transpose` rules, not `jvp` rules. See the documentation for more details!

**Reply by Rasmuskh** [reply]
Thanks, that is exactly what I was looking for!
